### PR TITLE
Special strip and pixel Clients to run Online on XeXe collisions

### PR DESCRIPTION
**Pull request by fioriNTU**
This PR is a first attempt to have running clients for Pixel and Strip dedicated to XeXe running.
For the time being the Online Tracking has been disabled (The HI community is interested mainly to low level quantities to be monitored Online). However I will try to include also the tracking and reiterate if needed.

These clients are no more crashing with the test runs, 304458 as an example.

### CODE REVIEW

File: DQM/Integration/python/clients/pixel_dqm_sourceclient-live_cfg.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import FWCore.ParameterSet.Config as cms
 
-from Configuration.StandardSequences.Eras import eras
+#from Configuration.StandardSequences.Eras import eras
 
-process = cms.Process("PIXELDQMLIVE", eras.Run2_2017)
+process = cms.Process("PIXELDQMLIVE")
```
Comment (makortel) [inline DQM/Integration/python/clients/pixel_dqm_sourceclient-live_cfg.py]: Why remove the era?

Comment (fioriNTU) [reply DQM/Integration/python/clients/pixel_dqm_sourceclient-live_cfg.py]: at the moment there no suitable era and if I keep this one I get immediately a crash

Anyway, is the era duing anything in Tracker local reco?

Comment (makortel) [reply DQM/Integration/python/clients/pixel_dqm_sourceclient-live_cfg.py]: > at the moment there no suitable era and if I keep this one I get immediately a crash

Ok (interesting, what kind of crash?)

> Anyway, is the era duing anything in Tracker local reco?

Yes, it enables the "phase1 mode" of RawToDigi
https://github.com/cms-sw/cmssw/blob/26f67af4327f839c41bb86555eb523cd9737ed19/EventFilter/SiPixelRawToDigi/python/SiPixelRawToDigi_cfi.py#L22
and tunes clusterizer parameters
https://github.com/cms-sw/cmssw/blob/26f67af4327f839c41bb86555eb523cd9737ed19/RecoLocalTracker/SiPixelClusterizer/python/SiPixelClusterizer_cfi.py#L32-L40

**Comment by davidlange6** [comment]
> On Oct 9, 2017, at 3:08 PM, fioriNTU <notifications@github.com> wrote:
> 
> @fioriNTU commented on this pull request.
> 
> In DQM/Integration/python/clients/pixel_dqm_sourceclient-live_cfg.py:
> 
> >  
> -process = cms.Process("PIXELDQMLIVE", eras.Run2_2017)
> +process = cms.Process("PIXELDQMLIVE")
> 
> at the moment there no suitable era and if I keep this one I get immediately a crash
> 
> Anyway, is the era duing anything in Tracker local reco?

yes - you do have a new detector after all

for example
https://cmssdt.cern.ch/lxr/source/RecoLocalTracker/SiPixelClusterizer/python/SiPixelClusterizer_cfi.py#0032

File: DQM/Integration/python/clients/pixel_dqm_sourceclient-live_cfg.py (hunk 1 above)
Comment (boudoul) [reply DQM/Integration/python/clients/pixel_dqm_sourceclient-live_cfg.py]: right was about to give the same, the era is needed for the phase1 pixel (even before the tracking)

Comment (fioriNTU) [reply DQM/Integration/python/clients/pixel_dqm_sourceclient-live_cfg.py]: ok @boudoul and @makortel , I will add it back as soon as we have a "working era", is it used also for strip local reco?

Comment (fioriNTU) [reply DQM/Integration/python/clients/pixel_dqm_sourceclient-live_cfg.py]: well, looking at the changes are so few that I could even add them manually ...

**Comment by boudoul** [comment]
could you please share the message of the crash when using the era ?

**Comment by fioriNTU** [comment]
@boudoul  the message is the same that Tomas circulated:
Exception Message:
python encountered the error: <type 'exceptions.NameError'>
name 'gtEvmDigis' is not defined

I don't really get where gtEvmDigis enter the processing, it is for sure not in our sequence

File: DQM/Integration/python/clients/pixel_dqm_sourceclient-live_cfg.py [hunk 2]
```diff
@@ -102,9 +107,22 @@
...
-
-
+    process.castorDigis.InputLabel = cms.InputTag("rawDataRepacker")
+    process.csctfDigis.producer = cms.InputTag("rawDataRepacker")
+    process.dttfDigis.DTTF_FED_Source = cms.InputTag("rawDataRepacker")
+    process.ecalDigis.InputLabel = cms.InputTag("rawDataRepacker")
+    process.ecalPreshowerDigis.sourceTag = cms.InputTag("rawDataRepacker")
+    process.gctDigis.inputLabel = cms.InputTag("rawDataRepacker")
+    process.gtDigis.DaqGtInputTag = cms.InputTag("rawDataRepacker")
+    process.gtEvmDigis.EvmGtInputTag = cms.InputTag("rawDataRepacker")
```
Comment (davidlange6) [inline DQM/Integration/python/clients/pixel_dqm_sourceclient-live_cfg.py]: hi @fioriNTU - this line is presumably the cause of your problem... you might consider using this helper function to help with future maintenance. 

https://github.com/cms-sw/cmssw/blob/master/FWCore/ParameterSet/python/MassReplace.py#L119

**Comment by makortel** [comment]
@fioriNTU Can you run `python <yourconfig>` instead of cmsRun and post the full traceback?

File: DQM/Integration/python/clients/pixel_dqm_sourceclient-live_cfg.py [hunk 3]
```diff
@@ -102,9 +107,34 @@
...
+    #replace the effect of era in local reco
+    process.siPixelDigis.UsePhase1=True
+    process.siPixelClusters.VCaltoElectronGain      = cms.int32(47)   # L2-4: 47 +- 4.7 
+    process.siPixelClusters.VCaltoElectronGain_L1   = cms.int32(50)   # L1:   49.6 +- 2.6 
+    process.siPixelClusters.VCaltoElectronOffset    = cms.int32(-60)  # L2-4: -60 +- 130 
+    process.siPixelClusters.VCaltoElectronOffset_L1 = cms.int32(-670) # L1:   -670 +- 220 
+    process.siPixelClusters.ChannelThreshold        = cms.int32(10) 
+    process.siPixelClusters.SeedThreshold           = cms.int32(1000) 
+    process.siPixelClusters.ClusterThreshold        = cms.int32(4000) 
+    process.siPixelClusters.ClusterThreshold_L1     = cms.int32(2000)
```
Comment (makortel) [inline DQM/Integration/python/clients/pixel_dqm_sourceclient-live_cfg.py]: These should probably be `siPixelClustersPreSplitting`.

File: DQM/Integration/python/clients/pixel_dqm_sourceclient-live_cfg.py (hunk 2 above)
Comment (fioriNTU) [reply DQM/Integration/python/clients/pixel_dqm_sourceclient-live_cfg.py]: @davidlange6 even commenting that line I get the same crash. Thanks for the information about the helper function.

File: DQM/Integration/python/clients/pixel_dqm_sourceclient-live_cfg.py (hunk 3 above)
Comment (fioriNTU) [reply DQM/Integration/python/clients/pixel_dqm_sourceclient-live_cfg.py]: here is the printout with python <pixel client> runkey=hi_run

Source: cms.Source("DQMStreamerReader",
    SelectEvents = cms.untracked.vstring('*'),
    delayMillis = cms.untracked.uint32(500),
    deleteDatFiles = cms.untracked.bool(False),
    endOfRunKills = cms.untracked.bool(True),
    minEventsPerLumi = cms.untracked.int32(1),
    nextLumiTimeoutMillis = cms.untracked.int32(240000),
    runInputDir = cms.untracked.string('/tmp'),
    runNumber = cms.untracked.uint32(111),
    scanOnce = cms.untracked.bool(False),
    skipFirstLumis = cms.untracked.bool(False),
    streamLabel = cms.untracked.string('streamDQM')
)

Loaded configuration file from: []
dqmRunConfig: cms.PSet(
    collectorHost = cms.untracked.string('dqm-c2d07-29.cms'),
    collectorPort = cms.untracked.int32(9190),
    type = cms.untracked.string('userarea')
)
Traceback (most recent call last):
  File "pixel_dqm_sourceclient-live_cfg.py", line 92, in <module>
    process.load("Configuration.StandardSequences.RawToDigi_Repacked_cff")
  File "/opt/offline/slc7_amd64_gcc630/cms/cmssw/CMSSW_9_2_12/python/FWCore/ParameterSet/Config.py", line 582, in load
    module = __import__(moduleName)
  File "/opt/offline/slc7_amd64_gcc630/cms/cmssw/CMSSW_9_2_12/python/Configuration/StandardSequences/RawToDigi_Repacked_cff.py", line 10, in <module>
    gtEvmDigis.EvmGtInputTag = 'rawDataRepacker'
NameError: name 'gtEvmDigis' is not defined

**Comment by davidlange6** [comment]
it seems you are drilling into configuration that is not otherwise being tested/debugged as there is no HI run this year... (and L1 has completely changed since the last one). I would suggest staying with the pp reco as is being done in the tier0.. (aside from the rawDataRepacker change to all the raw data input tags) [but I think this was already discussed last week]

**Comment by boudoul** [comment]
Hi @threus and @dmitrijus , some questions :  is tracker the only one using the hi -key or this is also true for all subsystems ?

**Comment by dmitrijus** [comment]
@boudoul I don't know what you mean by "using".

The "hi" key is propagated to all the online DQM clients, but it is up to them if they use it for configuration.
Most clients simply ignore this key...

**Comment by boudoul** [comment]
thanks @dmitrijus , I see. I was probably confused myself to due my own ignorance of the machinery.

**Comment by fioriNTU** [comment]
No more useful after XeXe, closing

**Pull request closed without merging**